api/src/main/java/javax/enterprise/inject/spi/BeanManager.java

```diff
@@ -406,14 +408,25 @@
      * @throws ContextNotActiveException if there is no active context object for the given scope
      * @throws IllegalArgumentException if there is more than one active context object for the given scope
      */
-    public Context getContext(Class<? extends Annotation> scopeType);
+    Context getContext(Class<? extends Annotation> scopeType);
+
+    /**
+     * Creates a new {@link ManageableContext} that represents the given built-in scope.  This ManageableContext is not
+     * yet active, it must be activated when ready to use.  Upon activation, you may use it to retrieve beans from
+     * the context.
+     *
+     * @param scopeType the  {@linkplain javax.enterprise.context scope}
+     * @return the {@linkplain ManageableContext context object}
+     * @throws IllegalArgumentException if the given scopeType is not a valid built in scope.
+     */
+    ManageableContext createManageableContext(Class<? extends Annotation> scopeType);
```

Comment:
The point is that a context is active with respect to the current thread and so it's not necessary to create a new managable context instance per invocation. It's the same as with registering regular contexts - an extension developer registers a single instance of context, see also `javax.enterprise.inject.spi.AfterBeanDiscovery.addContext(Context)`. Of course, it would be a different case if the EG decides to specify `ManageableContext` outside the context hierarchy (and possibly rename it).